sky/skylet/events.py

```diff
@@ -130,32 +130,48 @@ def _stop_cluster(self, autostop_config):
+            # cache_stopped_nodes.
             self._replace_yaml_for_stopping(self._ray_yaml_path,
                                             autostop_config.down)
 
-            # `ray up` is required to reset the upscaling speed and min/max
-            # workers. Otherwise, `ray down --workers-only` will continuously
-            # scale down and up.
-            logger.info('Running ray up.')
-            script = (cloud_vm_ray_backend.
-                      write_ray_up_script_with_patched_launch_hash_fn(
-                          self._ray_yaml_path,
-                          ray_up_kwargs={'restart_only': True}))
-            subprocess.run(
-                [sys.executable, script],
-                check=True,
-                # Use environment variables to disable the ray usage collection
-                # (to avoid overheads and potential issues with the usage)
-                # as sdk does not take the argument for disabling the usage
-                # collection.
-                env=dict(os.environ, RAY_USAGE_STATS_ENABLED='0'),
-            )
-
-            logger.info('Running ray down.')
-            # Stop the workers first to avoid orphan workers.
-            subprocess.run(
-                ['ray', 'down', '-y', '--workers-only', self._ray_yaml_path],
-                check=True)
+            # We do "initial ray up + ray down --workers-only" only for
+            # multinode clusters for two reasons: (1) optimization; they are
+            # not needed for single-node; (2) for single-node SSO clusters, we
+            # have seen a weird bug where user image's /etc/profile.d may
+            # contain the two AWS env vars, and so they take effect in the
+            # initial 'ray up', throwing a RuntimeError when some private VPC
+            # is not found (since it only exists in the assumed role, not in
+            # the custome principal set by the env vars).
+            if is_cluster_multinode:
+                # `ray up` is required to reset the upscaling speed and min/max
+                # workers. Otherwise, `ray down --workers-only` will
+                # continuously scale down and up.
+                logger.info('Running ray up.')
+                script = (cloud_vm_ray_backend.
+                          write_ray_up_script_with_patched_launch_hash_fn(
+                              self._ray_yaml_path,
+                              ray_up_kwargs={'restart_only': True}))
+                subprocess.run(
+                    [sys.executable, script],
+                    check=True,
+                    # Use environment variables to disable the ray usage
+                    # collection (to avoid overheads and potential issues with
+                    # the usage) as sdk does not take the argument for
+                    # disabling the usage collection.
+                    env=dict(os.environ, RAY_USAGE_STATS_ENABLED='0'),
```

Comment:
Can we set the `AWS_ACCESS_KEY_ID=""` here as well to be more robust to any AWS credential set in the environment by the user? It should work as our cluster should have been assumed with skypilot-v1 role.

Reply:
(From PR desc) I considered unsetting AWS_ACCESS_KEY_ID / AWS_SECRET_ACCESS_KEY in first ray up action, but that seems overly AWS specific logic in events.py.

Reply:
Sorry, I missed the description. Another way for that is we just use `env=dict(RAY_USAGE_STATS_ENABLED='0')` to avoid any potential interruption from the environment variable (may need double check). I still feel like it would be generally better to avoid this kind of user-specific environment affecting our skylet service.

Reply:
Removing os.environ works. This is a good call as it'll fix the bug for multinode clusters too. Updated.